src/sentry/statistical_detectors/issue_platform_adapter.py

```diff
@@ -0,0 +1,74 @@
+import hashlib
+import uuid
+from datetime import datetime, timezone
+from typing import List
+
+from sentry.issues.grouptype import PerformanceDurationRegressionGroupType
+from sentry.issues.issue_occurrence import IssueEvidence, IssueOccurrence
+from sentry.issues.producer import produce_occurrence_to_kafka
+from sentry.seer.utils import BreakpointData
+from sentry.utils import metrics
+
+
+def fingerprint_regression(transaction, automatic_detection=True):
+    prehashed_fingerprint = (
+        f"p95_transaction_duration_regression-{transaction}"
+        if automatic_detection
+        else f"p95_transaction_duration_regression-{transaction}-experiment"
+    )
+    return hashlib.sha1((prehashed_fingerprint).encode()).hexdigest()
+
+
+class Regression(BreakpointData):
```

Comment:
Why does Regression inherit from `BreakpointData`? That's conceptually confusing. It would make more sense if it contained breakpoint data rather than *being* breakpoint data

Reply:
`BreakpointData` contained `project` as a project slug for legacy reasons but we needed project_id for issue platform. Now I'm passing project_id as project

Reply:
That's not quite what I'm getting at. The structure should be:

```python3
class Regression:
    breakpointdata: BreakpointData
    project_id: int
    # whatever else we will need in the future
```

inheritance implies something IS a thing
A regression is a concept which INCLUDES breakpoint data, there's no need to inherit from it here even if at this time they are mostly the same thing

Reply:
I'm no longer creating Regression. Just reusing BreakpointData

Reply:
ah I see